src/topology/config/mod.rs

```diff
@@ -194,6 +194,13 @@ pub struct TransformOuter {
 
 #[typetag::serde(tag = "type")]
 pub trait TransformConfig: core::fmt::Debug {
+    /// Allows a transform configuration to expand itself into multiple "child"
+    /// transformations to replace it. This allows a transform to act as a macro
+    /// for various patterns.
+    fn expand(&mut self) -> Option<IndexMap<String, Box<dyn TransformConfig>>> {
```

Comment:
This is really cool! I've been thinking about a way to make an expansion/optimization pass on configs before actually building them and this seems like a nice and simple way to start experimenting with that.